sno/working_copy/postgis.py

```diff
@@ -51,19 +65,24 @@ def __init__(self, repo, uri):
             raise ValueError("Expecting postgresql://")
 
         url_path = url.path
-        self.schema = None
-        if url.path:
-            path_parts = url.path[1:].split("/")
-            if len(path_parts) > 2:
-                raise ValueError("Expecting postgresql://[host]/dbname/schema")
-            elif len(path_parts) == 2:
-                self.schema = path_parts[1]
-                url_path = f"/{path_parts[0]}"
-            else:
-                self.schema = self._default_schema(repo)
+        path_parts = url_path[1:].split("/", 3) if url_path else []
+        if len(path_parts) != 2:
+            raise ValueError("Expecting postgresql://[host]/dbname/schema")
+        self.schema = path_parts[1]
+
+        url_path = f"/{path_parts[0]}"
+        url_query = f"options=--search_path%3D{self.schema},public"
```

Comment:
there could be lots of schemas in the search path, defined on a user + db level. PostGIS and its functions could be in a separate schema, etc.

I think the safest is for us to refer to fully-qualified table paths everywhere and don't touch `search_path`

Reply:
Fair enough: this was a hack to avoid having to specify the schema in the test-helper functions. But, it doesn't make sense to mess up the production code for that reason only